File: problems/algorithms/2161/Cpp/solution_filtering.cpp
class Solution
{
public:
	vector<int> pivotArray(vector<int> &nums, int pivot)
	{
		vector<int> result, higher;
		int equal = 0;
		for (int num : nums)
		{
			if (num == pivot)
			{
				equal += 1;
			}
			else if (num < pivot)
			{
				result.push_back(num);
			}
			else
			{
				higher.push_back(num);
			}
		}
		result.insert(std::end(result), equal, pivot);
		result.insert(std::end(result), std::begin(higher), std::end(higher));
		return result;
	}
};

File: problems/algorithms/2161/Rust/solution_filtering.rs
impl Solution {
    pub fn pivot_array(nums: Vec<i32>, pivot: i32) -> Vec<i32> {
        let mut result = vec![];
        let mut equal = 0;
        let mut higher = vec![];
        for &num in &nums {
            if num == pivot {
                equal += 1;
            } else if num < pivot {
                result.push(num);
            } else {
                higher.push(num);
            }
        }
        for _ in 0..equal {
            result.push(pivot);
        }
        result.append(&mut higher);
        result
    }
}